### ISSUE TITLE
HUD conflicts with the latest version of Modern Casual preloader

### ISSUE THREAD

**Issue by spectywecty** [mcolour/colourhud#18]
HUD elements related to the objective like payload progress and cap points are completely gone if used along with the latest version of MCP.
![20240302181028_1](https://github.com/mcolour/colourhud/assets/116623617/b043456d-960f-4119-9a1f-ef5f123c402e)
With MCP
![20240302221609_1](https://github.com/mcolour/colourhud/assets/116623617/e07e1cf0-7069-49bc-a208-671f009164c6)
With no other mods besides mastercom and ColourHUD

**Comment by mcolour** [comment]
Fixed it

**Comment by spectywecty** [comment]
sweet! Thanks for the fix. I'm sure I wouldn't understand why the objective elements weren't working with MCP but how does colourhud load itself differently from most other huds? I used a couple huds as a replacement and they didn't seem to have similar issues like colour.

**Comment by mcolour** [comment]
When I added the sixense text writing support for the hud (the technology that allows you to edit the hud in-game), I also added the file with hud's default settings, so that colourhud would not be broken if the user started up the game with this hud for the first time and hasn't customized it yet. What I failed to notice is that `hudlayout.res`  (main file, responsible for hud rendering) has the same element names as in other `.res` files. I thought that naming the "namespace" of the default settings as `settings` would be fine, as TF2 handles the namespace problem by itself. However, MCP somehow makes TF2 skip this auto-handling process and just doesn't render some elements, like the payload progress bar. So I had to create a separate default settings file just for `hudlayout.res` with the namespace `"Resource/HudLayout.res"`.
It's pretty much source spaghetti.
